components/Login/Login.tsx

```diff
@@ -1,41 +1,64 @@
 import React, {useState} from 'react';
-import {Text, View} from 'react-native';
+import {KeyboardTypeOptions, Text, View} from 'react-native';
 import Form from '../Form/Form';
 import constants from '../../utils/constants';
 import {loginUser} from '../../api/user';
 import {ButtonType, Data} from '../../types/CommonTypes';
+import {useMutation} from 'react-query';
+import {useDispatch} from 'react-redux';
+import {logIn} from '../../redux/reducers/User';
 
 import style from './style';
 import globalStyle from '../../assets/styles/globalStyle';
 
 const Login: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const signInUser = useMutation({
+    mutationFn: async (data: {email: string; password: string}) => {
+      return await loginUser(data.email, data.password);
+    },
+    onError: async (error: string) => {
+      setError(error);
```

Comment:
You can probably destructure `{error}` from the hook `useMutation` and get rid of this state.